src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/CollectionModelBinderProvider.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Microsoft.Extensions.Internal;
+
+namespace Microsoft.AspNetCore.Mvc.ModelBinding.Binders
+{
+    /// <summary>
+    /// An <see cref="IModelBinderProvider"/> for <see cref="ICollection{T}"/>.
+    /// </summary>
+    public class CollectionModelBinderProvider : IModelBinderProvider
+    {
+        /// <inheritdoc />
+        public IModelBinder Create(ModelBinderProviderContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            var modelType = context.Metadata.ModelType;
+            
+            // Arrays are handled by another binder.
+            if (modelType.IsArray)
+            {
+                return null;
+            }
+
+            // If the model type is ICollection<> then we can call its Add method, so we can always support it.
+            var collectionType = ClosedGenericMatcher.ExtractGenericInterface(modelType, typeof(ICollection<>));
+            if (collectionType != null)
+            {
+                var elementType = collectionType.GenericTypeArguments[0];
+                var elementBinder = context.CreateBinder(context.MetadataProvider.GetMetadataForType(elementType));
+
+                var binderType = typeof(CollectionModelBinder<>).MakeGenericType(collectionType.GenericTypeArguments);
+                return (IModelBinder)Activator.CreateInstance(binderType, elementBinder);
+            }
+
+            // If the model type is IEnumerable<> then we need to know if we can assign a List<> to it, since
+            // that's what we would create.
+            var enumerableType = ClosedGenericMatcher.ExtractGenericInterface(modelType, typeof(IEnumerable<>));
+            if (enumerableType != null)
+            {
+                var listType = typeof(List<>).MakeGenericType(enumerableType.GenericTypeArguments);
+                if (modelType.GetTypeInfo().IsAssignableFrom(listType.GetTypeInfo()))
```

Comment:
Check whether the property is readonly here. Anything but `ICollection<>` is useful `iff` we can actually do the assignment.